### PR TITLE
notifications: clear notifications on pressing button

### PR DESCRIPTION
**Pull request by rcrdlbl**
see: #1549.
the "visibilitychange" listener was waiting for a visibility change to run, which doesn't make sense if the page is already open and someone hits the button.

also fixes missing key console warning in `Notification`

### CODE REVIEW

**Review by patosullivan — APPROVED** [review]
LGTM, works locally.

File: ui/vite.config.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default ({ mode }) => {
 
   return defineConfig({
     base: mode === 'mock' ? undefined : '/apps/grid/',
-    server: mode === 'mock' ? undefined : { https: true },
+    server: mode === 'mock' || mode === 'development' ? undefined : { https: true },
```
Comment (patosullivan) [inline ui/vite.config.ts:18]: Can we get rid of this? We do need https locally for testing push notifications in development.

**Review by patosullivan — CHANGES_REQUESTED** [review]
Actually this changes the dev server setup in a way we don't want, see my comment.

File: ui/vite.config.ts (hunk 1 above)
Comment (rcrdlbl) [reply ui/vite.config.ts]: just reverted this.

**Review by patosullivan — APPROVED** [review]

**Pull request merged**